migrate/cockroach.go

```diff
@@ -0,0 +1,183 @@
+package migrate
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/mattes/migrate"
+	"github.com/mattes/migrate/database/cockroachdb"
+	_ "github.com/mattes/migrate/source/file"
+
+	_ "github.com/lib/pq"
+)
+
+type Cockroach struct {
+	db *sql.DB
+
+	MigrationsPath string
+}
+
+type CockroachOptions struct {
+	*ConnectionOptions
+
+	MigrationTable string
+	Secure         bool
+	SSL            *CockroachSSL
+}
+
+type CockroachSSL struct {
+	CertPath, KeyPath, Mode, RootCert string
+}
+
+const (
+	DefaultHost           = "localhost"
+	DefaultUser           = "root"
+	DefaultPort           = "26257"
+	DefaultDatabase       = "service"
+	DefaultMigrationTable = "schema_migrations"
+)
+
+// Connection string concatenates the CockroachOptions down to a string,
+// applying defaults to options that were not set ready to be used in a
+// connection to the database
+func (co *CockroachOptions) ConnectionString() (string, error) {
+	// Prevent panics and just return the exact default upon nil options
+	if co.ConnectionOptions == nil {
+		return fmt.Sprintf(
+			"postgresql://%s@%s:%s/%s?sslmode=disable&x-migrations-table=%s",
+			DefaultUser, DefaultHost, DefaultPort, DefaultDatabase, DefaultMigrationTable,
+		), nil
+	}
+
+	// Start a string builder for the connection string
+	var conn strings.Builder
+	conn.WriteString("postgresql://")
+	// User
+	if len(co.User) != 0 {
+		conn.WriteString(co.User)
+	} else {
+		conn.WriteString(DefaultUser)
+	}
+
+	conn.WriteString("@")
+
+	// Host
+	if len(co.Host) != 0 {
+		conn.WriteString(co.Host)
+	} else {
+		conn.WriteString(DefaultHost)
+	}
+
+	conn.WriteString(":")
+
+	// Port
+	if co.Port != 0 {
+		conn.WriteString(strconv.Itoa(co.Port))
+	} else {
+		conn.WriteString(DefaultPort)
+	}
+
+	conn.WriteString("/")
+
+	if len(co.Database) != 0 {
+		conn.WriteString(co.Database)
+	} else {
+		conn.WriteString(DefaultDatabase)
+	}
+
+	// Set connection security
+	if co.Secure {
```

Comment:
What's the plan for getting hold of the certificate and key when targeting a remote cluster?

Reply:
According to [migrate](https://github.com/mattes/migrate/tree/master/database/cockroachdb)   the options take a path to a PEM encoded cert on the machine. The only issue I see standing is classic PKI distribution but seems a bigger challenge than just this package